cljfmt/src/cljfmt/core.cljc

```diff
@@ -341,6 +346,35 @@
   ([form indents alias-map]
    (transform form edit-all should-indent? #(indent-line % indents alias-map))))
 
+(defn map-key? [zloc]
+  (and
+    (z/map? (z/up zloc))
+    (even? (index-of zloc))
+    (not (uneval-node? zloc))
+    (not (whitespace-or-comment? zloc))))
+
+(defn preceded-by-line-break? [zloc]
+  (loop [previous (zip/left zloc)]
+    (cond
+      (nil? previous)
+      false
+      (line-break? previous)
+      true
+      (whitespace-or-comment? previous)
+      (recur (zip/left previous))
+      :else
+      false)))
+
+(defn map-key-without-line-break? [zloc]
+  (and (map-key? zloc) (not (preceded-by-line-break? zloc))))
+
+(defn split-keypairs-over-multiple-lines [form]
+  (transform
```

Comment:
This should use standard Clojure indentation, and could be more concisely written:

```clojure
(defn- insert-newline-left [zloc]
  (zip/insert-left zloc (n/newlines 1)))

(defn split-keypairs-over-multiple-lines [form]
  (transform form edit-all map-key-without-line-break? insert-newline-left))
```